fastlane/fastlane
# [action] add --enable-code-coverage option to generate-xcodeproj command for the spm action

### Checklist

### Motivation and Context

The spm action does not currently include the option to enable code coverage for generated Xcode projects. Using an xcconfig file would enable code coverage support, but not in the generated scheme.

Including this option would allow Swift Packages to be built and tested, while also gathering coverage, in CI/CD lanes without the need for manually generated or modified Xcode projects. Without this, a manual modification to the generated project's scheme would be required to gather coverage.


### Description


Added the `enable_code_coverage` option to the spm action.
This option would only be included when the command is generate-xcodeproject.

### Testing Steps


Use the spm action to generate an Xcode project and check the "Gather code coverage" option is enabled within the project's main scheme.
